unstructured/partition/text.py

```diff
@@ -26,52 +27,125 @@
 )
 
 
-def split_by_paragraph(content: str, max_partition: Optional[int] = 1500) -> List[str]:
-    paragraphs = re.split(PARAGRAPH_PATTERN, content)
-    if max_partition is None:
-        return paragraphs
+def _split_in_half_at_breakpoint(
+    content: str,
+    breakpoint: str = " ",
+) -> List[str]:
+    """Splits a segment of content at the breakpoint closest to the middle"""
+    mid = len(content) // 2
+    for i in range(len(content) // 2):
+        if content[mid + i] == breakpoint:
+            mid += i
+            break
+        elif content[mid - i] == breakpoint:
+            mid += -i
+            break
 
-    split_paragraphs = []
-    for paragraph in paragraphs:
-        split_paragraphs.extend(
-            _split_to_fit_max_content(paragraph, max_partition=max_partition),
-        )
-    return split_paragraphs
+    return [content[:mid].rstrip(), content[mid:].lstrip()]
 
 
 def _split_content_size_n(content: str, n: int) -> List[str]:
-    """Splits a string into chunks that are at most size n."""
+    """Splits a section of content into chunks that are at most
+    size n without breaking apart words."""
     segments = []
-    for i in range(0, len(content), n):
-        segment = content[i : i + n]  # noqa: E203
-        segments.append(segment)
+    if len(content) < n * 2:
+        segments = list(_split_in_half_at_breakpoint(content))
+    else:
+        segments = textwrap.wrap(content, width=n)
     return segments
 
 
-def _split_to_fit_max_content(content: str, max_partition: int = 1500) -> List[str]:
+def split_content_to_fit_min_max(
+    content: str,
+    max_partition: Optional[int] = 1500,
+    min_partition: Optional[int] = 0,
+) -> List[str]:
     """Splits a section of content so that all of the elements fit into the
-    max partition window."""
+    max/min partition window."""
     sentences = sent_tokenize(content)
-    num_sentences = len(sentences)
-
+    # import pdb; pdb.set_trace()
     chunks = []
-    chunk = ""
-
-    for i, sentence in enumerate(sentences):
-        if len(sentence) > max_partition:
-            chunks.extend(_split_content_size_n(sentence, n=max_partition))
-
-        if len(chunk + " " + sentence) > max_partition:
-            chunks.append(chunk)
-            chunk = sentence
+    tmp = ""
```

Comment:
Could you change this back to `chunk` (or `tmp_chunk` would be fine too) to make the role of the variable more clear?